src/explore-education-statistics-admin/src/pages/methodology/edit-methodology/content/context/useMethodologyActions.ts

```diff
@@ -0,0 +1,247 @@
+import methodologyService from '@admin/services/methodology/methodologyService';
+import permissionsService from '@admin/services/permissions/service';
+import { Dictionary } from '@admin/types';
+import { useCallback } from 'react';
+import { ContentBlockPostModel } from 'src/services/release/edit-release/content/types';
+import { useMethodologyDispatch } from './MethodologyContext';
+import { ContentSectionKeys } from './MethodologyContextActionTypes';
+
+export default function useMethodologyActions() {
+  const dispatch = useMethodologyDispatch();
+  const annexes = 'annexes';
+
+  const getMethodologyContent = useCallback(
+    async (methodologyId: string) => {
+      dispatch({ type: 'CLEAR_STATE' });
+      const methodology = await methodologyService.getMethodologyContent(
+        methodologyId,
+      );
+      const canUpdateMethodology = await permissionsService.canUpdateMethodology(
+        methodologyId,
+      );
+      dispatch({
+        type: 'SET_STATE',
+        payload: {
+          methodology,
+          canUpdateMethodology,
+        },
+      });
+    },
+    [dispatch],
+  );
+
+  async function deleteContentSectionBlock({
+    methodologyId,
+    sectionId,
+    blockId,
+    isAnnex = false,
+  }: {
+    methodologyId: string;
+    sectionId: string;
+    blockId: string;
+    isAnnex?: boolean;
+  }) {
+    await methodologyService.deleteContentSectionBlock(
+      methodologyId,
+      sectionId,
+      blockId,
+      isAnnex,
+    );
+    dispatch({
+      type: 'REMOVE_BLOCK_FROM_SECTION',
+      payload: {
+        meta: { sectionId, blockId, sectionKey: isAnnex ? annexes : 'content' },
```

Comment:
As suggested earlier, we could just pass a `sectionKey` through instead of having to do all of these boolean checks